src/librustc/ty/context.rs

```diff
@@ -2338,8 +2338,12 @@ impl<'tcx> TyCtxt<'tcx> {
     }
 
     #[inline]
-    pub fn mk_generator_witness(self, types: ty::Binder<&'tcx List<Ty<'tcx>>>) -> Ty<'tcx> {
-        self.mk_ty(GeneratorWitness(types))
+    pub fn mk_generator_witness(
+        self,
+        types: ty::Binder<&'tcx List<Ty<'tcx>>>,
+        region_outlives: &'tcx List<Predicate<'tcx>>,
```

Comment:
@Aaron1011 Should `List<RegionOutlivesPredicate>` be wrapped inside `Binder` like types in witness?

Reply:
@csmoe: I think so (assuming it also contains late-bound regions that were 'anonymized' from the original regions).